### PR TITLE
add ghcr publish

### PR DESCRIPTION
**Pull request by ksurl**
PR for https://github.com/haugene/docker-transmission-openvpn/issues/1633

# PREREQUISITES  
* create github personal access token with permissions for read/write packages as seen [here](https://docs.github.com/en/packages/guides/migrating-to-github-container-registry-for-docker-images)
* add the token as a Circle CI variable named GITHUB_PASS

# POST MERGE  
* after first build has been published, package needs to be made public (defaults to private)

### CODE REVIEW

**Comment by ksurl** [comment]
I was thinking about adding docker layer caching but it seems a bit more of rewrite since this workflow is using manual commands to do the build instead of the template actions from Circle CI.

**Comment by ksurl** [comment]
looks like it failed. did the github token and circle ci secret get created?

**Comment by pkishino** [comment]
yes, I added it..checking details now

**Comment by pkishino** [comment]
hmm, says 403 forbidden.. the login succeeds..
https://app.circleci.com/pipelines/github/haugene/docker-transmission-openvpn/467/workflows/a9eb381f-04c2-4881-907e-3ab370cac015/jobs/1119

**Comment by pkishino** [comment]
I'll revert this for now..

**Comment by ksurl** [comment]
I can't see the output of the workflow since I am not in the organization. As far as I can tell the buildx syntax is correct. I just added a 2nd `-t` for the new tag. My guess is it currently does not support pushing to multiple registries in 1 go.

I could try either adding a 2nd step for ghcr. it should rebuild instantly since the local cache should be identical. or rebuild the workflow using the templates that should work with it since it would loop for all tags

**Comment by pkishino** [comment]
here is the last part:
```
#41 pushing layers
#41 pushing layers 4.3s done
#41 pushing manifest for docker.io/*******/transmission-openvpn:latest
#41 pushing manifest for docker.io/*******/transmission-openvpn:latest 2.6s done
#41 pushing layers 0.3s done
#41 ERROR: unexpected status: 403 Forbidden
------
 > exporting to image:
------
failed to solve: rpc error: code = Unknown desc = unexpected status: 403 Forbidden
```

**Comment by ksurl** [comment]
maybe missed one more step on the github account side, github container registry needs to be enabled in feature preview

https://docs.github.com/en/packages/guides/enabling-improved-container-support

**Comment by pkishino** [comment]
ok, let me try again

**Comment by pkishino** [comment]
well..na..let see when @haugene can check this..perhaps has to do with project permissions

**Comment by pkishino** [comment]
I think all that is needed is for @haugene to create a PAT for the DOCKER_USER account..won't work with mine I believe..

**Comment by ksurl** [comment]
Oh right. I thought that was implied since it would be pushing to his account. Is haugene a regular user or an organization in github? I don’t think collaborators on a repo would have access. The github container registry treats the images repo as a separate thing from the git repo at the account level unlike the old docker package repo that was tied to the git repo

**Comment by pkishino** [comment]
yeah, he is the project owner

**Comment by ksurl** [comment]
@haugene please take a look and create the access token when you get a chance. Thanks.

**Comment by ksurl** [comment]
any update @haugene ?
